docs/utilities.rst

```diff
@@ -0,0 +1,58 @@
+.. cpp:namespace:: nanobind
+
+.. _utilities:
+
+Utilities
+==========
+
+.. _utilities_eval:
+
+Evaluating Python expressions from strings
+==========================================
+
+nanobind provides the ``eval`` and ``exec`` functions to evaluate Python
+expressions and statements. The following example illustrates how they can be
+used.
+
+.. code-block:: cpp
+
+    // At beginning of file
+    #include <nanobind/eval.h>
+
+    ...
+
+    // Evaluate in scope of main module
+    nb::object scope = nb::module_::import_("__main__").attr("__dict__");
+
+    // Evaluate an isolated expression
+    int result = nb::eval("my_variable + 10", scope).cast<int>();
+
+    // Evaluate a sequence of statements
+    nb::exec(
+        "print('Hello')\n"
+        "print('world!');",
+        scope);
+
+C++11 raw string literals are also supported and quite handy for this purpose.
+The only requirement is that the first statement must be on a new line following
+the raw string delimiter ``R"(``, ensuring all lines have common leading indent:
+
+.. code-block:: cpp
+
+    nb::exec(R"(
+        x = get_answer()
+        if x == 42:
+            print('Hello World!')
+        else:
+            print('Bye!')
+        )", scope
+    );
+
+.. note::
+
+    `eval` accepts a template parameter that describes how the string/file
```

Comment:
Can you add sphinx prefixes (e.g. `:py:func:..` to referenced function and enumerations values to ensure that those are linked to the right parts of the API documentation?

Reply:
(I don't see the change for this resolved comment)

Reply:
I'd only changed it for the first occurence of eval/exec; fixed now for that line too.